lib/typeahead/index.js

```diff
@@ -95,15 +100,23 @@ var Typeahead = React.createClass({
 
   getOptionsForValue: function (value, options) {
     if (this._shouldSkipSearch(value)) {
-      return [];
+      return {
+        visible: [],
+        truncated: false
+      };
     }
 
     var searchOptions = this._generateSearchFunction();
     var result = searchOptions(value, options);
+    var resultLength = result.length;
     if (this.props.maxVisible) {
       result = result.slice(0, this.props.maxVisible);
     }
-    return result;
+
+    return {
```

Comment:
It could be that this is a good time to change the internal state of react-typeahead so it stores the current (complete) search result set.

Right now most of the data is being garbage collected almost immediately.
We could change `visible` to `searchResults` and in the render method, only display up to `maxVisible`

It's a pretty in-depth refactor that would require a number of unit-tests, but it would make this a lot better. It also clears up https://github.com/fmoo/react-typeahead/pull/194 because `visible` isn't really the right word for "first `maxVisible` search results with metadata"

I honestly might want to drop something like this into the original PR. But it's a really good next step.